wf-galleon/WFLY-19021-Stability_In_Provisioning.adoc

```diff
@@ -0,0 +1,194 @@
+* https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven Plugin]
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin[WildFly Bootable JAR]
+* https://github.com/wildfly-extras/prospero[Prospero]
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+Stability becomes a Galleon notion that applies to feature-pack, feature/feature_param and package.
+There is a 1 to 1 mapping with WildFly stability.
+The stability levels are: `default` < `community` < `preview` < `experimental`
+
+=== Requirements at feature-pack build time
+
+==== feature-pack minimum stability level
+
+A minimum stability level can be set at feature-pack build time. If no stability level is specified, the `default` level is used.
+
+Stability level is set thanks to a new option of the WildFly Galleon Plugins Maven plugin: 
+`<minimum-stability>default|community|preview|experimental</minimum-stability>`
+
+Feature-specs and packages at a lower stability level than is specified will not be included in the feature pack.
+
+==== JBoss Modules modules stability level
+
+A stability level can be set in `module.xml` thanks to the property `jboss.stability`. 
+Possible value being `default|community|preview|experimental`. If no stability is specified, `default` is assumed.
+
+==== Implementation notes
+
+At build time, this minimum stability level is used in 2 cases:
+
+* To start the embedded server that generates the Galleon features at the correct stability level (allowing the features to be discovered in the server).
+* To filter-out packages at a stability level lower than the minimum-stability level. Such packages are not packaged inside the feature-pack.
+
+==== Stability for WildFly feature-packs
+
+Expected minimum stability levels for WildFly feature-packs:
+
+* wildfly-ee-galleon-pack: `experimental`
+* wildfly-galleon-pack: `experimental`
+* wildfly-preview-feature-pack: `experimental`
+
+=== Requirements at provisioning time
+
+Provisioning time follows some rules to properly handle stability and avoid features/packages of wrong stability being provisioned.
+
+* The feature-pack contained minimum stability level is used to constrain the set of provisioned features. Any features/packages at a lower stability level contained in the 
+feature-pack (if any), are not provisioned.
+* By default, the minimum stability level of each feature-pack is used when provisioning its own content. 
+* Each feature-pack present in the provisioning configuration can have a different minimum stability level. 
+The minimum stability of each feature-pack applies to its own content and doesn't leak into other feature-packs.
+* A user can specify a stability-level, by mean of the `stability-level` Galleon option. 
```

Comment:
Oops, my previous requested change here had it's own typo!

s/mean/means

Reply:
Fixed.